app/src/ui/clone-repository/clone-repository.tsx

```diff
@@ -269,9 +269,9 @@ export class CloneRepository extends React.Component<
   private updateAndValidatePath = async (path: string) => {
     this.setState({ path })
 
-    const doesDirectoryExist = await this.doesPathExist(path)
+    const pathValidation = await this.validateEmptyFolder(path)
 
-    if (doesDirectoryExist) {
+    if (pathValidation != null) {
```

Comment:
Should we use `pathValidation` on the next line?

```ts
if (pathValidation != null) {
  this.setState({ error: pathValidation })
} else {
  this.setState({ error: null })
}
```

Reply:
I had considered this as it seemed like a pretty natural change. My decision against doing so came down the other possible errors not seeming fit to display to a user.

However, this change would make even more sense since we will no longer be returning the raw errors via the validation function.
___
Even a step further, if we're willing to set the state to `pathValidation` directly, we could even skip the `if/else` entirely since `pathValidation` will always either be an `Error` or `null`.

If we don't need any additional validation after the other changes then this would be a quite nice clean up.

Reply:
> Even a step further, if we're willing to set the state to `pathValidation` directly, we could even skip the `if/else` entirely since `pathValidation` will always either be an `Error` or null.

I like this idea